File: scrape_accurate_data.py
import requests
import json

def get_accurate_stations():
    print("Fetching accurate stations from eRail...")
    url = "https://erail.in/data/stations.json"
    headers = {
        "User-Agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36"
    }
    try:
        response = requests.get(url, headers=headers)
        if response.status_code == 200:
            raw_data = response.json()
            cleaned_stations = [{"code": st[0], "name": st[1]} for st in raw_data if len(st) >= 2]
            with open("railsetu_stations.json", "w", encoding="utf-8") as f:
                json.dump(cleaned_stations, f, ensure_ascii=False, indent=4)
            print(f"Success! {len(cleaned_stations)} accurate stations saved.")
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    get_accurate_stations()

File: scrape_accurate_data.js
import fs from 'fs';

async function getAccurateStations() {
    console.log("Fetching accurate stations from eRail...");
    const url = "https://erail.in/data/stations.json";
    const headers = {
        "User-Agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36"
    };

    try {
        const response = await fetch(url, { headers });
        if (response.ok) {
            const rawData = await response.json();
            const cleanedStations = rawData
                .filter(st => st.length >= 2)
                .map(st => ({
                    code: st[0],
                    name: st[1]
                }));
            
            fs.writeFileSync("railsetu_stations.json", JSON.stringify(cleanedStations, null, 4), "utf-8");
            console.log(`Success! ${cleanedStations.length} accurate stations saved.`);
        } else {
            console.error(`Error: HTTP ${response.status}`);
        }
    } catch (e) {
        console.error(`Error: ${e}`);
    }
}

getAccurateStations();
